File: src/omnia/data/structure/observable/writable/WritableObservableSet.java
package omnia.data.structure.observable.writable;

import omnia.data.contract.Writable;
import omnia.data.structure.mutable.MutableSet;
import omnia.data.structure.observable.ObservableSet;

public interface WritableObservableSet<E> extends MutableSet<E>, ObservableSet<E>, Writable<ObservableSet<E>> {

  static <E> WritableObservableSet<E> create() {
    return new WritableObservableSetImpl<>();
  }
}

File: src/omnia/data/structure/observable/writable/WritableObservableSet.kt
package omnia.data.structure.observable.writable

import omnia.data.contract.Writable
import omnia.data.structure.mutable.MutableSet
import omnia.data.structure.observable.ObservableSet

interface WritableObservableSet<E : Any> : MutableSet<E>, ObservableSet<E>, Writable<ObservableSet<E>> {
  companion object {

    @JvmStatic
    fun <E : Any> create(): WritableObservableSet<E> {
      return WritableObservableSetImpl()
    }
  }
}